ui/src/app/app.tsx

```diff
@@ -67,18 +67,25 @@ async function isExpiredSSO() {
 
 requests.onError.subscribe(async err => {
     if (err.status === 401) {
-        if (!history.location.pathname.startsWith('/login')) {
-            // Query for basehref and remove trailing /.
-            // If basehref is the default `/` it will become an empty string.
-            const basehref = document
-                .querySelector('head > base')
-                .getAttribute('href')
-                .replace(/\/$/, '');
-            if (await isExpiredSSO()) {
-                window.location.href = `${basehref}/auth/login?return_url=${encodeURIComponent(location.href)}`;
-            } else {
-                history.push(`${basehref}/login?return_url=${encodeURIComponent(location.href)}`);
-            }
+        if (history.location.pathname.startsWith('/login')) {
+            return;
+        }
+
+        const isSSO = await isExpiredSSO();
```

Comment:
tests?

Reply:
I think this is only possible to test is in e2e test ( Even if we  could test app.tsx ) . Will do it once we get to UI e2e tests

Reply:
if you approve #2610  you can do this without maybe?